Atomic.md

```diff
@@ -0,0 +1,270 @@
+Atomic placeholder operations in VFSForGit on MacOS
+===================================================
+
+A potential hole in our current design is in our creation of
+placeholders. Today, when creating a placeholder directory, at a high
+level, we go through the following workflow:
+
+-   Create placeholder folder/file in the virtualization root
+
+-   'Mark up' the file with the proper flags/xattrs
+
+-   (If creating a file) Update the mode of the file
+
+This could result in scenarios where a partial placeholder appears in
+the virtualization root. The full set of failure modes is documented
+below:
+
+WritePlaceholderDirectory:
+
+-   mkdir() fails to create the placeholder directory
+
+    -   The enumeration fails as a directory wasn't created. This should
+         be retriable as a second attempt should see the IsEmpty bit on
+         the parent folder and call WritePlaceholderDirectory again. No
+         partial placeholders were left in the parent directory.
+
+-   We crash between mkdir() and InitializeEmptyPlaceholder
+
+    -   Placeholder folder appears to be not in virtualization root and
```

Comment:
> appears to be not in virtualization root

To confirm, in today's implementation the directory would not be flagged as inside the root?